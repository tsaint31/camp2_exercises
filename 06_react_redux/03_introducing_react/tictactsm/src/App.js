import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


function hasWinner(state) {
  const WINNING_COORDINATES = [
    [{letter: "a", digit: "0"}, {letter: "a", digit: "1"}, {letter: "a", digit: "2"}],
    [{letter: "b", digit: "0"}, {letter: "b", digit: "1"}, {letter: "b", digit: "2"}],
    [{letter: "c", digit: "0"}, {letter: "c", digit: "1"}, {letter: "c", digit: "2"}],
    [{letter: "a", digit: "0"}, {letter: "b", digit: "1"}, {letter: "c", digit: "2"}],
    [{letter: "a", digit: "2"}, {letter: "b", digit: "1"}, {letter: "c", digit: "0"}],
    [{letter: "a", digit: "0"}, {letter: "b", digit: "0"}, {letter: "c", digit: "0"}],
    [{letter: "a", digit: "1"}, {letter: "b", digit: "1"}, {letter: "c", digit: "1"}],
    [{letter: "a", digit: "2"}, {letter: "b", digit: "2"}, {letter: "c", digit: "2"}]
  ];

  const isWinningLine = (line) => {
    const pattern =
      line
      .map((coordinate) => state[coordinate.letter][coordinate.digit])
      .join("");

    return pattern === "XXX" || pattern === "OOO";
  }

  return WINNING_COORDINATES.some(isWinningLine);
}

function RenderCell(props) {
  if (props.cell === "_") {
    return <button className="square" onClick={() => props.handleInput()} >{props.cell}</button>;
  } else {
    return <button className="square"  >{props.cell}</button>;
  }
}

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      status: { a: Array(3).fill("_"), b: Array(3).fill("_"),c: Array(3).fill("_")},
      xIsNext: true
    };
    this.handleInput=this.handleInput.bind(this)
  }

  handleInput(i,letter) {
    if (hasWinner(this.state.status)) {
      return;
    }
    let modifystatus = this.state.status;
    modifystatus[letter][i]=this.state.xIsNext ? 'X' : 'O';
    this.setState({
      status: modifystatus,
      xIsNext: !this.state.xIsNext
    });
  }

  renderRow = (letter) => {
    const cells = this.state.status[letter];
    return (cells.map((cell,i) => <RenderCell key={i} letter={letter} cell={cell} handleInput={() => this.handleInput(i,letter)} />));
  }

  board() {
    const letters = Object.keys(this.state.status);
    const rows = letters.map(this.renderRow);
    return (
        <div>
        <div className="board-row">
          {rows[0]}
        </div>
        <div className="board-row">
          {rows[1]}
        </div>
        <div className="board-row">
          {rows[2]}
        </div>
        </div>
    );
  }

  restart() {
    let modifystatus = { a: Array(3).fill("_"), b: Array(3).fill("_"),c: Array(3).fill("_")};
    this.setState({
      status: modifystatus,
    });
  }

  render() {
    const winner = hasWinner(this.state.status);
    let stat;
    if (winner && this.state.xIsNext) {
      stat = 'THE WINNER IS 0';
    }
    else if (winner) {
      stat = 'THE WINNER is X';
    }
    else {
      stat = 'Next player: ' + (this.state.xIsNext ? 'X' : 'O');
    }

    return (
      <div>
            <div className="App">
              <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <h1 className="App-title">TIC TAC TOE</h1>
              </header>
              <p className="App-intro">
              Enjoy your tictactoe </p>
            </div>
      <div className="status"><button onClick={() => this.restart()} >RESTART</button></div>
      <div className="status">{stat}</div>
          {this.board()}
      </div>

    );
  }
}

export default App;
