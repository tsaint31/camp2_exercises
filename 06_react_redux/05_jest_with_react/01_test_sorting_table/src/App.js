import React, { Component } from "react";
import _ from "underscore";

function Row(props) {
  return (
    <tr>
      <td>{props.decathlon_id}</td>
      <td>{props.title}</td>
      <td>{props.price}</td>
    </tr>
  );
}

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lines: this.props.lines,
      sortedBy: "decathlon_id",
      reverseSort: false
    };
  }

  filter(filterBy) {
    if (this.state.sortedBy === filterBy) {
      this.setState({reverseSort: !this.state.reverseSort});
    } else {
      this.setState({
        sortedBy: filterBy,
        reverseSort: false
      });
    }
  }

  render() {
    let sortedLines = _.sortBy(this.state.lines, this.state.sortedBy);
    if (this.state.reverseSort) {
      sortedLines.reverse();
    }
    return (
      <table>
        <thead>
          <tr>
            <th id="th1" onClick={() => this.filter("decathlon_id")}>ID</th>
            <th  onClick={() => this.filter("title")}>Title</th>
            <th  onClick={() => this.filter("price")}>Price</th>
          </tr>
        </thead>
        {sortedLines.map(Row)}
      </table>
    );
  }
}

export default App;
export {Row};
