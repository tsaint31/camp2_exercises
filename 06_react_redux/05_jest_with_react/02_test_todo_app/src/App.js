import React, { Component } from "react";

let lastId = 1;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      newTask: "",
      tasks: []
    };

    this.addTask = this.addTask.bind(this);
    this.deleteTask = this.deleteTask.bind(this);
    this.handleNewTask = this.handleNewTask.bind(this);
    this.updateTask = this.updateTask.bind(this);
  }

  addTask(event) {
    event.preventDefault();
    const newTask = {
      id: lastId,
      label: this.state.newTask,
      completed: false
    };
    lastId++;
    this.setState({
      tasks: [...this.state.tasks, newTask],
      newTask: ""
    });
  }

  deleteTask(taskId) {
    const newTasks = this.state.tasks.filter(task => task.id !== taskId);
    console.log(newTasks)
    this.setState({tasks: newTasks});
  }

  handleNewTask(event) {
    this.setState({
      newTask: event.target.value
    });
  }

  updateTask(taskId) {
    const newTasks = this.state.tasks.map(task => {
      if (task.id === taskId) {
        task.completed = !task.completed;
      }
      return task;
    });
    this.setState({tasks: newTasks});
  }

  render() {
    return (
      <div className="Todo">
        <div className="new-todo">
          <form onSubmit={this.addTask}>
            <input type="text" value={this.state.newTask} onChange={this.handleNewTask} />
            <button id="addbutton" type="submit">Add</button>
          </form>
        </div>
        <ul className="todos">
          {this.state.tasks.map(task =>
            <li key={task.id} className={task.completed ? "completed" : "task"}>
              <span>{task.label}</span>
              <input
                type="checkbox"
                value={task.completed}
                onChange={() => this.updateTask(task.id)}
              />
              { task.completed
                ? <input type="button" onClick={() => this.deleteTask(task.id)} value="delete" />
                : null
              }
            </li>
          )}
        </ul>
      </div>
    )
  }
}

export default App;
