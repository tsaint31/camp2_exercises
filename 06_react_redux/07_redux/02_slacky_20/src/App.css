html, body, #root {
  height: 100%;
}

.App {
  text-align: center;
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
  height: 100%;
  align-items: stretch;
  align-content: stretch;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 50px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.Login {
  padding: 20px;
}

.Login div {
  padding: 20px;
}

.Chat {
  flex: 2 1;
  display:flex;
  flex-flow: column;
  justify-content: space-between;
  padding-top: 5px;
  align-items: stretch;
  align-content: stretch;
  max-height: calc(100vh - 50px)
}

.Chat-messages {
  flex: 1 1 auto;
  overflow: auto;
  max-height: calc(100% - 50px)
}

.Chat-form {
  height: 50px;
  background: #ccc;
  display: flex;
}
.Chat-form form {
  display: flex;
  width: 100%;
}

.Chat-form input {
  flex: 1 0;
}

.Chat-form button {
  height: 100%;
  margin: 0;
  padding: 20px;
  background-color: #007bff;
  color: white;
  border: none;
  -webkit-appearance: none;
}

.message-container {
  width: 80%;
  text-align: left;
}

.author {
  font-weight: bold;
  padding: 5px 10px;
}
