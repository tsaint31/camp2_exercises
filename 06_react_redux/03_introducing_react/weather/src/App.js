import React, { Component } from 'react';
import './App.css';
import logo from './logo.svg';
import weatherByCityName from './weather.js';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isFetching: false,
      current: "",
      temp:""
    };
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({isFetching: true}, () => {
      weatherByCityName(this.state.current)
      .then(output => {
        this.setState({
          isFetching: false,
          temp: output
        });
      });
    })
  }

  saveCity = evt => {
    this.setState({
      current: evt.target.value
    });
  }

  render() {
    const loading = (<h1>LOADING...</h1>);

    return (
      <div>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">WEATHER</h1>
          </header>
          <p className="App">
              Find the weather of your city </p>
        </div>
        <form onSubmit={this.handleSubmit}>
          <label>
            Please enter your city
            <input
              placeholder={"task"}
              type="text"
              value={this.state.current}
              onChange={this.saveCity}
            />
          </label>
        </form>
        {this.state.isFetching && loading}
        {!this.state.isFetching && (
          <div>
            <div className="App">
              Here is the temperature of your town
            </div>
            <div className="App-intro">
              {this.state.temp}
            </div>
          </div>)
        }
      </div>
    );
  }
}

export default App;
