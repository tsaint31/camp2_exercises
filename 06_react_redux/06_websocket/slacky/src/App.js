import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


function Rows(props){
  let classformat="column1"
  if (props.name===props.slack.user)
  {
    classformat="column"
  }
  return (
    <tr>
      <td>
        {props.slack.slack}
      </td>
      <td className={classformat}>
        {props.slack.user}
      </td>
    </tr>
  );
}

function isJson(str) {
    try {
        JSON.parse(str);
    } catch (e) {
        return false;
    }
    return true;
}

function Userlist(props){
  let classformat="column1"
  if (props.name===props.user)
  {
    classformat="column"
  }
  return (
    <tr>
      <td className={classformat}>
        {props.user}
      </td>
    </tr>
  );
}

class App extends Component {
  constructor(props) {
    super(props);
    this.ws= new WebSocket("ws://localhost:4000")
    this.state = {
      name: "",
      current: "" ,
      currentslack: "" ,
      isFetching: false ,
      slack: [],
      users:0,
      userslist:[]
    };
  }

  componentDidMount() {
  // What to do when we receive a message?
  this.ws.onmessage = (event) => {
    console.log(event);
    if (!isNaN(event.data)) {
      console.log(event);
      this.setState({
        users:event.data
      })
    }
    else if (isJson(event.data)) {
      console.log(event);
      console.log(event.data);
      let newslacklist = this.state.slack;
      newslacklist.push(JSON.parse(event.data));
      this.setState({
        slack:newslacklist
      });
    }
    else
    {
      const newuser=this.state.userslist;
      newuser.push(event.data);
      this.setState({
        userslist:newuser
      });
    }
  };

  this.ws.onsend = (event) => {
    console.log(event);
  };
  // Alert the server that the client is gone
  window.addEventListener("beforeunload", () => this.ws.send("CLOSE"));
}

  fetchData = evt => {
    evt.preventDefault();
    let newName = this.state.current;
    this.ws.send(newName);
    this.setState({
      name: newName,
      isFetching:true,
    });
  };

  fetchSlack = evt => {
    evt.preventDefault();
    const newSlack = {
      slack: "",
      user: this.state.name
    };
    newSlack.slack = this.state.currentslack;
    this.ws.send(JSON.stringify(newSlack));
  };

  saveName = evt => {
    this.setState({
      current: evt.target.value
    });
  };

  saveSlack = evt1 => {
    this.setState({
      currentslack: evt1.target.value
    });
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to Slacky</h1>
        </header>
        <p className="App-intro">
        Enjoy your slacky (nb of users connected : {this.state.users}) </p>

        {!this.state.isFetching && (
          <div>
          <label>
            Please enter your name
            <input
              placeholder="name"
              type="text"
              value={this.state.current}
              onChange={this.saveName}
            />
          </label>
        <button onClick={this.fetchData} > Submit </button>
        </div>)}
        {this.state.isFetching && (
          <div>
          <h1 className="App"> Welcome {this.state.name} to the slacky line </h1>
          <label>
            Please enter your slacky
            <input
              placeholder="slack"
              type="text"
              value={this.state.currentslack}
              onChange={this.saveSlack}
            />
          </label>
        <button onClick={this.fetchSlack} > Submit </button>
        <table>
          <thead>
            <tr>
              <th> List of Slackies</th>
              <th> From</th>
            </tr>
          </thead>
          <tbody>{this.state.slack.map((slack,i) => <Rows key={i} slack={slack} name={this.state.name}/>)}</tbody>
        </table>
        <table>
          <thead>
            <tr>
              <th> List of Slackies USERS</th>
            </tr>
          </thead>
          <tbody>{this.state.userslist.map((user,i) => <Userlist key={i} user={user} name={this.state.name}/>)}</tbody>
        </table>
        </div>)
        }
      </div>
    );
  }
}

export default App;
