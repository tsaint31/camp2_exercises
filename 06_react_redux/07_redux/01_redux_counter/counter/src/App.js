import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from "react-redux";

class App extends Component {
render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to Counter</h1>
        </header>
        <p className="App-intro">
        Enjoy your counter </p>
        <div>
          <button onClick={this.props.decrement}> - </button>
          <span>{this.props.counter}</span>
          <button onClick={this.props.increment}> + </button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    counter: state.counter
  };
}

function mapDispatchToProps(dispatch) {
  return {
    decrement: () => dispatch({type: "DECREMENT"}),
    increment: () => dispatch({type: "INCREMENT"})
  }
}

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default ConnectedApp;
