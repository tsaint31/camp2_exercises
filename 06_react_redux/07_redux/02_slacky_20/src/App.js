import React, { Component } from "react";
import "./App.css";
import Login from "./Login";
import Chat from "./Chat";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
      users:[]
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket("ws://localhost:8080");
  }
  componentDidMount() {
    // Listen for messages
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          this.setState({ messages: message.data });
          return;

      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName });
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = message => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: message
      })
    );
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Slacky</h1>
        </header>

        {this.state.userName ? (
          <Chat sendMessage={this.sendMessage} messages={this.state.messages} />
        ) : (
          <Login handleUserName={this.handleUserName} />
        )}
      </div>
    );
  }
}

export default App;
