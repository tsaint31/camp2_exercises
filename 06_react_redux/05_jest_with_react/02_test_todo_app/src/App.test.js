import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wapiti from "wapiti";


configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <App> my app </App>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test(' check text', () => {
  const app = shallow(
    <App />);
  expect(app.text()).toEqual("Add");
  app.setState({ tasks: ["exo1","exo2"] });

  expect(app.find("li").length).toBe(2);

  app.setState({ newTask: "exo3" });
  app.find("form").simulate('Submit', { preventDefault() {}});
  expect(app.find("li").length).toBe(3);

});

test("should display a counter starting at 0", () => {
  expect.assertions(1);
  return Wapiti.goto("http://localhost:3000")
    .capture(() => document.querySelector("#addbutton").textContent)
    .run()
    .then(result => {
      expect(result).toBe("Add");
    });
});
