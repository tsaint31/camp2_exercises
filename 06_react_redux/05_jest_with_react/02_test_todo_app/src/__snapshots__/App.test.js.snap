// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="Todo"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        id="addbutton"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  />
</div>
`;
